upgrade helix-front to Angular 6.1

### Is your feature request related to a problem? Please describe.

<!--- A clear and concise description of what the problem is. -->

Two acute problems motivate upgrading the web app `helix-front` to the latest version of the Angular UI framework (Angular 13 at this writing).

1) node-sass dependency doesn't support M1 MacBooks.

2) Other dependencies have security vulnerabilities. 

for example:

```
> Notes: Vulnerability found: ansi-regex is vulnerable to Inefficient Regular Expression Complexity remediation: Upgrade ansi-regex from 4.1.1 to 5.0.1 to fix the vulnerability.
```

### Describe the solution you'd like
<!--- A clear and concise description of what you want to happen. -->

Upgrade `helix-front` to Angular 13.  Perform other dependency upgrades necessary in order to make helix-front easy to maintain and contribute to going forward.

Specifically, replace **node-sass** with **dart-sass**, since **node-sass** is deprecated and **dart-sass** does support modern M1 / arm64 Macs. 

### Additional context
<!--- Add any other context or screenshots about the feature request here. -->

This change should also fix #1574 #1573

`npm run build` fails with this log. 

I'll pick up here next time. This upgrade to Angular 6.1 is _almost_ done 🤞 

```py
npm run build

> helix-front@1.2.1 build /helix/helix-front
> rm -rf dist && mkdir dist && ng build --aot --prod && tsc -p server


Date: 2022-05-05T00:45:41.224Z
Hash: 68266db706cfe5dc9f9c
Time: 14299ms
chunk {0} runtime.06daa30a2963fa413676.js (runtime) 1.44 kB [entry] [rendered]
chunk {1} main.8b8040c86f701990c967.js (main) 128 bytes [initial] [rendered]
chunk {2} polyfills.63d6937766727c57970f.js (polyfills) 130 bytes [initial] [rendered]
chunk {3} styles.8d5b34660fde16b98bd3.css (styles) 135 kB [initial] [rendered]

ERROR in node_modules/rxjs/ReplaySubject.d.ts(1,15): error TS2307: Cannot find module 'rxjs-compat/ReplaySubject'.
node_modules/angulartics2/dist/core/angulartics2.d.ts(1,10): error TS2305: Module '"/helix/helix-front/node_modules/rxjs/ReplaySubject"' has no exported member 'ReplaySubject'.
client/app/shared/node-viewer/node-viewer.component.ts(236,17): error TS2339: Property 'value' does not exist on type 'number | ListFieldObject | ((...items: ListFieldObject[]) => number) | ((searchElement: ListField...'.
  Property 'value' does not exist on type 'number'.
client/app/shared/node-viewer/node-viewer.component.ts(240,35): error TS2345: Argument of type 'number | ListFieldObject | ((...items: ListFieldObject[]) => number) | ((searchElement: ListField...' is not assignable to parameter of type 'ListFieldObject'.
  Type 'number' is not assignable to type 'ListFieldObject'.
client/app/shared/node-viewer/node-viewer.component.ts(247,27): error TS2339: Property 'value' does not exist on type 'number | MapFieldObject | ((...items: MapFieldObject[]) => number) | ((searchElement: MapFieldObj...'.
  Property 'value' does not exist on type 'number'.
client/app/shared/node-viewer/node-viewer.component.ts(273,43): error TS2339: Property 'value' does not exist on type 'number | ListFieldObject | ((...items: ListFieldObject[]) => number) | ((searchElement: ListField...'.
  Property 'value' does not exist on type 'number'.
client/app/shared/node-viewer/node-viewer.component.ts(275,19): error TS2339: Property 'value' does not exist on type 'number | ListFieldObject | ((...items: ListFieldObject[]) => number) | ((searchElement: ListField...'.
  Property 'value' does not exist on type 'number'.
client/app/shared/node-viewer/node-viewer.component.ts(277,19): error TS2339: Property 'value' does not exist on type 'number | ListFieldObject | ((...items: ListFieldObject[]) => number) | ((searchElement: ListField...'.
  Property 'value' does not exist on type 'number'.
client/app/shared/node-viewer/node-viewer.component.ts(279,35): error TS2345: Argument of type 'number | ListFieldObject | ((...items: ListFieldObject[]) => number) | ((searchElement: ListField...' is not assignable to parameter of type 'ListFieldObject'.
  Type 'number' is not assignable to type 'ListFieldObject'.
client/app/shared/node-viewer/node-viewer.component.ts(289,27): error TS2339: Property 'value' does not exist on type 'number | MapFieldObject | ((...items: MapFieldObject[]) => number) | ((searchElement: MapFieldObj...'.
  Property 'value' does not exist on type 'number'.
client/app/core/helix.service.ts(20,5): error TS2322: Type 'Observable<Object>' is not assignable to type 'Observable<boolean>'.
  Type 'Object' is not assignable to type 'boolean'.
client/app/resource/resource-list/resource-list.component.ts(62,12): error TS2339: Property 'map' does not exist on type 'Observable<Params>'.
client/app/controller/shared/controller.service.ts(12,8): error TS2339: Property 'map' does not exist on type 'Observable<any>'.
client/app/history/shared/history.service.ts(13,8): error TS2339: Property 'map' does not exist on type 'Observable<any>'.
client/app/history/shared/history.service.ts(19,8): error TS2339: Property 'map' does not exist on type 'Observable<any>'.
client/app/workflow/shared/job.service.ts(12,8): error TS2339: Property 'map' does not exist on type 'Observable<any>'.
client/app/dashboard/dashboard.component.ts(145,8): error TS2339: Property 'map' does not exist on type 'Observable<Params>'.
```

It looks like it did. Now we have some TypeScript errors to fix. Let's do that. 

```python
npm run build
# ...
ERROR in client/app/shared/node-viewer/node-viewer.component.ts(236,17): error TS2339: Property 'value' does not exist on type 'number | ListFieldObject | ((...items: ListFieldObject[]) => number) | ((searchElement: ListField...'.
  Property 'value' does not exist on type 'number'.
client/app/shared/node-viewer/node-viewer.component.ts(240,35): error TS2345: Argument of type 'number | ListFieldObject | ((...items: ListFieldObject[]) => number) | ((searchElement: ListField...' is not assignable to parameter of type 'ListFieldObject'.
  Type 'number' is not assignable to type 'ListFieldObject'.
client/app/shared/node-viewer/node-viewer.component.ts(247,27): error TS2339: Property 'value' does not exist on type 'number | MapFieldObject | ((...items: MapFieldObject[]) => number) | ((searchElement: MapFieldObj...'.
  Property 'value' does not exist on type 'number'.
client/app/shared/node-viewer/node-viewer.component.ts(273,43): error TS2339: Property 'value' does not exist on type 'number | ListFieldObject | ((...items: ListFieldObject[]) => number) | ((searchElement: ListField...'.
  Property 'value' does not exist on type 'number'.
client/app/shared/node-viewer/node-viewer.component.ts(275,19): error TS2339: Property 'value' does not exist on type 'number | ListFieldObject | ((...items: ListFieldObject[]) => number) | ((searchElement: ListField...'.
  Property 'value' does not exist on type 'number'.
client/app/shared/node-viewer/node-viewer.component.ts(277,19): error TS2339: Property 'value' does not exist on type 'number | ListFieldObject | ((...items: ListFieldObject[]) => number) | ((searchElement: ListField...'.
  Property 'value' does not exist on type 'number'.
client/app/shared/node-viewer/node-viewer.component.ts(279,35): error TS2345: Argument of type 'number | ListFieldObject | ((...items: ListFieldObject[]) => number) | ((searchElement: ListField...' is not assignable to parameter of type 'ListFieldObject'.
  Type 'number' is not assignable to type 'ListFieldObject'.
client/app/shared/node-viewer/node-viewer.component.ts(289,27): error TS2339: Property 'value' does not exist on type 'number | MapFieldObject | ((...items: MapFieldObject[]) => number) | ((searchElement: MapFieldObj...'.
  Property 'value' does not exist on type 'number'.
client/app/core/helix.service.ts(20,5): error TS2322: Type 'Observable<Object>' is not assignable to type 'Observable<boolean>'.
  Type 'Object' is not assignable to type 'boolean'.
client/app/resource/resource-list/resource-list.component.ts(62,12): error TS2339: Property 'map' does not exist on type 'Observable<Params>'.
client/app/controller/shared/controller.service.ts(12,8): error TS2339: Property 'map' does not exist on type 'Observable<any>'.
client/app/history/shared/history.service.ts(13,8): error TS2339: Property 'map' does not exist on type 'Observable<any>'.
client/app/history/shared/history.service.ts(19,8): error TS2339: Property 'map' does not exist on type 'Observable<any>'.
client/app/workflow/shared/job.service.ts(12,8): error TS2339: Property 'map' does not exist on type 'Observable<any>'.
client/app/dashboard/dashboard.component.ts(145,8): error TS2339: Property 'map' does not exist on type 'Observable<Params>'.
```

--- a/helix-front/client/app/app.module.ts
+++ b/helix-front/client/app/app.module.ts
@@ -1,7 +1,7 @@
 import { BrowserModule } from '@angular/platform-browser';
 import { NgModule } from '@angular/core';
 import { FormsModule } from '@angular/forms';
-import { HttpModule } from '@angular/http';
+import { HttpClientModule } from '@angular/common/http';
 
 import { Angulartics2Module, Angulartics2Piwik } from 'angulartics2';
 
@@ -26,7 +26,7 @@ import { DashboardModule } from './dashboard/dashboard.module';
   imports: [
     BrowserModule,
     FormsModule,
-    HttpModule,
+    HttpClientModule,
     AppRoutingModule,
     Angulartics2Module.forRoot([ Angulartics2Piwik ]),
     CoreModule,

--- a/helix-front/client/app/chooser/shared/chooser.service.spec.ts
+++ b/helix-front/client/app/chooser/shared/chooser.service.spec.ts
@@ -1,6 +1,6 @@
 import { TestBed, inject } from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
-import { HttpModule } from '@angular/http';
+import { HttpClientModule } from '@angular/common/http';
 
 import { ChooserService } from './chooser.service';
 
@@ -9,7 +9,7 @@ describe('ChooserService', () => {
     TestBed.configureTestingModule({
       imports: [
         RouterTestingModule,
-        HttpModule
+        HttpClientModule
       ],
       providers: [ChooserService]
     });

--- a/helix-front/client/app/cluster/cluster.module.ts
+++ b/helix-front/client/app/cluster/cluster.module.ts
@@ -1,6 +1,6 @@
 import { NgModule } from '@angular/core';
 import { CommonModule } from '@angular/common';
-import { HttpModule } from '@angular/http';
+import { HttpClientModule } from '@angular/common/http';
 
 import { SharedModule } from '../shared/shared.module';
 import { ClusterService } from './shared/cluster.service';
@@ -12,7 +12,7 @@ import { ClusterComponent } from './cluster.component';
 @NgModule({
   imports: [
     CommonModule,
-    HttpModule,
+    HttpClientModule,
     SharedModule
   ],
   declarations: [

--- a/helix-front/client/app/cluster/shared/cluster.service.spec.ts
+++ b/helix-front/client/app/cluster/shared/cluster.service.spec.ts
@@ -1,5 +1,5 @@
 import { TestBed, inject } from '@angular/core/testing';
-import { HttpModule } from '@angular/http';
+import { HttpClientModule } from '@angular/common/http';
 import { RouterTestingModule } from '@angular/router/testing';
 
 import { ClusterService } from './cluster.service';
@@ -7,7 +7,7 @@ import { ClusterService } from './cluster.service';
 describe('ClusterService', () => {
   beforeEach(() => {
     TestBed.configureTestingModule({
-      imports: [HttpModule, RouterTestingModule],
+      imports: [HttpClientModule, RouterTestingModule],
       providers: [ClusterService]
     });
   });

--- a/helix-front/client/app/configuration/shared/configuration.service.spec.ts
+++ b/helix-front/client/app/configuration/shared/configuration.service.spec.ts
@@ -1,13 +1,13 @@
 import { TestBed, inject } from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
-import { HttpModule } from '@angular/http';
+import { HttpClientModule } from '@angular/common/http';
 
 import { ConfigurationService } from './configuration.service';
 
 describe('ConfigurationService', () => {
   beforeEach(() => {
     TestBed.configureTestingModule({
-      imports: [HttpModule, RouterTestingModule],
+      imports: [HttpClientModule, RouterTestingModule],
       providers: [ConfigurationService]
     });
   });

--- a/helix-front/client/app/controller/controller-detail/controller-detail.component.spec.ts
+++ b/helix-front/client/app/controller/controller-detail/controller-detail.component.spec.ts
@@ -1,6 +1,6 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { NO_ERRORS_SCHEMA } from '@angular/core';
-import { HttpModule } from '@angular/http';
+import { HttpClientModule } from '@angular/common/http';
 import { RouterTestingModule } from '@angular/router/testing';
 
 import { ControllerDetailComponent } from './controller-detail.component';
@@ -12,7 +12,7 @@ describe('ControllerDetailComponent', () => {
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       imports: [
-        HttpModule,
+        HttpClientModule,
         RouterTestingModule
       ],
       declarations: [ ControllerDetailComponent ],

--- a/helix-front/client/app/controller/shared/controller.service.spec.ts
+++ b/helix-front/client/app/controller/shared/controller.service.spec.ts
@@ -1,5 +1,5 @@
 import { TestBed, inject } from '@angular/core/testing';
-import { HttpModule } from '@angular/http';
+import { HttpClientModule } from '@angular/common/http';
 import { RouterTestingModule } from '@angular/router/testing';
 
 import { ControllerService } from './controller.service';
@@ -7,7 +7,7 @@ import { ControllerService } from './controller.service';
 describe('ControllerService', () => {
   beforeEach(() => {
     TestBed.configureTestingModule({
-      imports: [HttpModule, RouterTestingModule],
+      imports: [HttpClientModule, RouterTestingModule],
       providers: [ControllerService]
     });
   });

--- a/helix-front/client/app/core/helix.service.spec.ts
+++ b/helix-front/client/app/core/helix.service.spec.ts
@@ -1,5 +1,5 @@
 import { TestBed, inject } from '@angular/core/testing';
-import { HttpModule } from '@angular/http';
+import { HttpClientModule } from '@angular/common/http';
 import { RouterTestingModule } from '@angular/router/testing';
 
 import { HelixService } from './helix.service';
@@ -7,7 +7,7 @@ import { HelixService } from './helix.service';
 describe('HelixService', () => {
   beforeEach(() => {
     TestBed.configureTestingModule({
-      imports: [HttpModule, RouterTestingModule],
+      imports: [HttpClientModule, RouterTestingModule],
       providers: [HelixService]
     });
   });

--- a/helix-front/client/app/core/helix.service.ts
+++ b/helix-front/client/app/core/helix.service.ts
@@ -1,5 +1,5 @@
 import { Injectable } from '@angular/core';
-import { Headers, Http, Response } from '@angular/http';
+import { HttpHeaders, HttpClient, HttpResponse } from '@angular/common/http';
 import { Router } from '@angular/router';
 import { Observable } from 'rxjs/Rx';
 
@@ -10,13 +10,12 @@ export class HelixService {
 
   constructor(
     protected router: Router,
-    private http: Http
+    private http: HttpClient
   ) { }
 
   public can(): Observable<boolean> {
     return this.http
       .get(`${ Settings.userAPI }/can`, { headers: this.getHeaders() })
-      .map(response => response.json())
       .catch(this.errorHandler);
   }
 
@@ -31,7 +30,6 @@ export class HelixService {
         `${Settings.helixAPI}${helix}${path}`,
         { headers: this.getHeaders() }
       )
-      .map(response => response.json())
       .catch(this.errorHandler);
   }
 
@@ -42,7 +40,6 @@ export class HelixService {
         data,
         { headers: this.getHeaders() }
       )
-      .map(response => response.text().trim() ? response.json() : '{}')
       .catch(this.errorHandler);
   }
 
@@ -53,7 +50,6 @@ export class HelixService {
         data,
         { headers: this.getHeaders() }
       )
-      .map(response => response.text().trim() ? response.json() : '{}')
       .catch(this.errorHandler);
   }
 
@@ -63,7 +59,6 @@ export class HelixService {
         `${Settings.helixAPI}${this.getHelixKey()}${path}`,
         { headers: this.getHeaders() }
       )
-      .map(response => response.text().trim() ? response.json() : '{}')
       .catch(this.errorHandler);
   }
 
@@ -73,7 +68,7 @@ export class HelixService {
   }
 
   protected getHeaders() {
-    let headers = new Headers();
+    let headers = new HttpHeaders();
     headers.append('Accept', 'application/json');
     headers.append('Content-Type', 'application/json');
     return headers;
@@ -84,12 +79,12 @@ export class HelixService {
 
     let message = error.message || 'Cannot reach Helix restful service.';
 
-    if (error instanceof Response) {
+    if (error instanceof HttpResponse) {
       if (error.status == 404) {
         // rest api throws 404 directly to app without any wrapper
         message = 'Not Found';
       } else {
-        message = error.text();
+        message = error;
         try {
           message = JSON.parse(message).error;
         } catch (e) {}

--- a/helix-front/client/app/core/user.service.ts
+++ b/helix-front/client/app/core/user.service.ts
@@ -1,5 +1,5 @@
 import { Injectable } from '@angular/core';
-import { Headers, Http, Response } from '@angular/http';
+import { HttpHeaders, HttpClient, HttpResponse } from '@angular/common/http';
 import { Router } from '@angular/router';
 import { Observable } from 'rxjs/Rx';
 
@@ -10,28 +10,26 @@ export class UserService {
 
   constructor(
     protected router: Router,
-    private http: Http
+    private http: HttpClient
   ) { }
 
-  public getCurrentUser(): Observable<string> {
+  public getCurrentUser(): Observable<string | object> {
     return this.http
       .get(`${ Settings.userAPI }/current`, { headers: this.getHeaders() })
-      .map(response => response.json())
       .catch(_ => _);
   }
 
-  public login(username: string, password: string): Observable<boolean> {
+  public login(username: string, password: string): Observable<object> {
     return this.http
       .post(
         `${ Settings.userAPI }/login`,
         { username: username, password: password },
         { headers: this.getHeaders() }
-      )
-      .map(response => response.json());
+      );
   }
 
   protected getHeaders() {
-    let headers = new Headers();
+    let headers = new HttpHeaders();
     headers.append('Accept', 'application/json');
     headers.append('Content-Type', 'application/json');
     return headers;

--- a/helix-front/client/app/history/history-list/history-list.component.spec.ts
+++ b/helix-front/client/app/history/history-list/history-list.component.spec.ts
@@ -1,6 +1,6 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { NO_ERRORS_SCHEMA } from '@angular/core';
-import { HttpModule } from '@angular/http';
+import { HttpClientModule } from '@angular/common/http';
 import { RouterTestingModule } from '@angular/router/testing';
 
 import { HistoryListComponent } from './history-list.component';
@@ -12,7 +12,7 @@ describe('HistoryListComponent', () => {
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       imports: [
-        HttpModule,
+        HttpClientModule,
         RouterTestingModule
       ],
       declarations: [ HistoryListComponent ],

--- a/helix-front/client/app/history/shared/history.service.spec.ts
+++ b/helix-front/client/app/history/shared/history.service.spec.ts
@@ -1,5 +1,5 @@
 import { TestBed, inject } from '@angular/core/testing';
-import { HttpModule } from '@angular/http';
+import { HttpClientModule } from '@angular/common/http';
 import { RouterTestingModule } from '@angular/router/testing';
 
 import { HistoryService } from './history.service';
@@ -7,7 +7,7 @@ import { HistoryService } from './history.service';
 describe('HistoryService', () => {
   beforeEach(() => {
     TestBed.configureTestingModule({
-      imports: [HttpModule, RouterTestingModule],
+      imports: [HttpClientModule, RouterTestingModule],
       providers: [HistoryService]
     });
   });

--- a/helix-front/client/app/instance/shared/instance.service.spec.ts
+++ b/helix-front/client/app/instance/shared/instance.service.spec.ts
@@ -1,5 +1,5 @@
 import { TestBed, inject } from '@angular/core/testing';
-import { HttpModule } from '@angular/http';
+import { HttpClientModule } from '@angular/common/http';
 import { RouterTestingModule } from '@angular/router/testing';
 
 import { InstanceService } from './instance.service';
@@ -7,7 +7,7 @@ import { InstanceService } from './instance.service';
 describe('InstanceService', () => {
   beforeEach(() => {
     TestBed.configureTestingModule({
-      imports: [HttpModule, RouterTestingModule],
+      imports: [HttpClientModule, RouterTestingModule],
       providers: [InstanceService]
     });
   });

--- a/helix-front/client/app/resource/resource-detail-for-instance/resource-detail-for-instance.component.spec.ts
+++ b/helix-front/client/app/resource/resource-detail-for-instance/resource-detail-for-instance.component.spec.ts
@@ -1,6 +1,6 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { NO_ERRORS_SCHEMA } from '@angular/core';
-import { HttpModule } from '@angular/http';
+import { HttpClientModule } from '@angular/common/http';
 import { RouterTestingModule } from '@angular/router/testing';
 
 import { ResourceService } from '../shared/resource.service';
@@ -12,7 +12,7 @@ describe('ResourceDetailForInstanceComponent', () => {
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      imports: [ HttpModule, RouterTestingModule ],
+      imports: [ HttpClientModule, RouterTestingModule ],
       declarations: [ ResourceDetailForInstanceComponent ],
       schemas: [
         /* avoid importing modules */

--- a/helix-front/client/app/resource/shared/resource.service.spec.ts
+++ b/helix-front/client/app/resource/shared/resource.service.spec.ts
@@ -1,5 +1,5 @@
 import { TestBed, inject } from '@angular/core/testing';
-import { HttpModule } from '@angular/http';
+import { HttpClientModule } from '@angular/common/http';
 import { RouterTestingModule } from '@angular/router/testing';
 
 import { ResourceService } from './resource.service';
@@ -7,7 +7,7 @@ import { ResourceService } from './resource.service';
 describe('ResourceService', () => {
   beforeEach(() => {
     TestBed.configureTestingModule({
-      imports: [HttpModule, RouterTestingModule],
+      imports: [HttpClientModule, RouterTestingModule],
       providers: [ResourceService]
     });
   });

--- a/helix-front/client/app/workflow/workflow-list/workflow-list.component.spec.ts
+++ b/helix-front/client/app/workflow/workflow-list/workflow-list.component.spec.ts
@@ -4,7 +4,7 @@ import { RouterTestingModule } from '@angular/router/testing';
 
 import { WorkflowListComponent } from './workflow-list.component';
 import { WorkflowService } from '../shared/workflow.service';
-import { HttpModule } from '@angular/http';
+import { HttpClientModule } from '@angular/common/http';
 
 describe('WorkflowListComponent', () => {
   let component: WorkflowListComponent;
@@ -12,7 +12,7 @@ describe('WorkflowListComponent', () => {
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      imports: [ HttpModule, RouterTestingModule ],
+      imports: [ HttpClientModule, RouterTestingModule ],
       declarations: [ WorkflowListComponent ],
       providers: [ WorkflowService ],
       schemas: [

--- a/helix-front/client/testing/testing.module.ts
+++ b/helix-front/client/testing/testing.module.ts
@@ -1,5 +1,5 @@
 import { NgModule } from '@angular/core';
-import { HttpModule } from '@angular/http';
+import { HttpClientModule } from '@angular/common/http';
 import { MaterialModule } from '../app/shared/material.module';
 import { RouterTestingModule } from '@angular/router/testing';
 import { NoopAnimationsModule } from '@angular/platform-browser/animations';
@@ -9,7 +9,7 @@ import { HelperServiceStub } from './stubs';
 
 @NgModule({
   imports: [
-    HttpModule,
+    HttpClientModule,
     MaterialModule,
     RouterTestingModule,
     NoopAnimationsModule
@@ -21,7 +21,7 @@ import { HelperServiceStub } from './stubs';
     }
   ],
   exports: [
-    HttpModule,
+    HttpClientModule,
     MaterialModule,
     RouterTestingModule,
     NoopAnimationsModule

--- a/helix-front/package.json
+++ b/helix-front/package.json
@@ -26,18 +26,19 @@
     "npm": "5.5.1"
   },
   "dependencies": {
-    "@angular/animations": "^5.1.1",
-    "@angular/cdk": "^5.0.1",
-    "@angular/common": "^5.1.1",
-    "@angular/compiler": "^5.1.1",
-    "@angular/core": "^5.1.1",
-    "@angular/flex-layout": "2.0.0-beta.12",
-    "@angular/forms": "^5.1.1",
-    "@angular/http": "^5.1.1",
-    "@angular/material": "^5.0.1",
-    "@angular/platform-browser": "^5.1.1",
-    "@angular/platform-browser-dynamic": "^5.1.1",
-    "@angular/router": "^5.1.1",
+    "@angular/animations": "^5.2.11",
+    "@angular/cdk": "^5.2.5",
+    "@angular/common": "^5.2.11",
+    "@angular/compiler": "^5.2.11",
+    "@angular/core": "^5.2.11",
+    "@angular/flex-layout": "^5.0.0-beta.14",
+    "@angular/forms": "^5.2.11",
+    "@angular/http": "^5.2.11",
+    "@angular/material": "^5.2.5",
+    "@angular/platform-browser": "^5.2.11",
+    "@angular/platform-browser-dynamic": "^5.2.11",
+    "@angular/platform-server": "^5.2.11",
+    "@angular/router": "^5.2.11",
     "@swimlane/ngx-charts": "^7.0.1",
     "@swimlane/ngx-datatable": "^11.1.7",
     "angulartics2": "^2.2.1",
@@ -58,13 +59,13 @@
     "ngx-vis": "^0.1.0",
     "node-sass": "4.5.3",
     "request": "^2.81.0",
-    "rxjs": "^5.5.5",
+    "rxjs": "^5.5.12",
     "vis": "^4.21.0",
     "zone.js": "^0.8.4"
   },
   "devDependencies": {
     "@angular/cli": "^1.6.1",
-    "@angular/compiler-cli": "^5.1.1",
+    "@angular/compiler-cli": "^5.2.11",
     "@types/hammerjs": "^2.0.34",
     "@types/jasmine": "2.5.38",
     "@types/lodash": "4.14.120",
@@ -85,6 +86,9 @@
     "protractor": "~5.1.0",
     "ts-node": "~2.0.0",
     "tslint": "^5.8.0",
-    "typescript": "2.4.2"
+    "typescript": "^2.4.2"
+  },
+  "volta": {
+    "node": "8.9.4"
   }
 }

